src/xhr.js

```diff
@@ -63,6 +79,46 @@ class Xhr {
 
 
 /**
+ * Normalized method name by uppercasing.
+ * @param {string|undefined} opt_method
+ * @return {string}
+ * @private
+ */
+export function normalizeMethod_(opt_method) {
+  if (opt_method === undefined) {
+    return 'GET';
+  }
+  return opt_method.toUpperCase();
```

Comment:
also removed `allowedMethods_` and any assertions. the reason why, is that when its the `native`
fetch method i'd be too late to do any assertions anyways or i would need to do it earlier (in `fetchJson`, but to do assertions there i would need to construct the opt_init object early too with its default properties)